data-pipeline/fetch_data.py

```
@@ -33,12 +33,47 @@
+        value = reading.get("value")
+        
+        if param not in self.VALIDATION_RANGES:
+            # If we don't have a specific range, assume valid for now
+            # or add a default check (e.g., value >= 0)
+            return value >= 0 if isinstance(value, (int, float)) else False
+            
+        min_val, max_val = self.VALIDATION_RANGES[param]
+        try:
+            val_float = float(value)
+            if not (min_val <= val_float <= max_val):
+                logger.warning(f"Invalid {param} value: {val_float}. Expected between {min_val} and {max_val}")
```

Ruff (0.14.14)

[warning] Logging statement uses f-string

(G004)